daemon/daemon_test.go

```diff
@@ -243,3 +254,54 @@ func (s *daemonSuite) TestAutherValidUser(c *check.C) {
 	c.Check(err, check.IsNil)
 	c.Check(user, check.DeepEquals, expectedUser.Authenticator())
 }
+
+type witnessAcceptListener struct {
+	net.Listener
+	accept chan struct{}
+}
+
+func (l *witnessAcceptListener) Accept() (net.Conn, error) {
+	close(l.accept)
+	return l.Listener.Accept()
+}
+
+func (s *daemonSuite) TestStartStop(c *check.C) {
+	d := newTestDaemon(c)
+	l, err := net.Listen("tcp", "127.0.0.1:0")
```

Comment:
hmm, I know we do this in a few places in the tests already, but this'll break on ipv6-only systems :-(
is it worth addressing it here, or should we do a separate branch to grab 'em all?